Infrastructure/BusinessLogic/BooksWorkflow.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Infrastructure.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Infrastructure.BusinessLogic
+{
+    public class BooksWorkflow : IBooksWorkflow
+    {
+        private ISalesRepository _salesRepository;
+        private IBooksRepository _booksRepository;
+
+        public BooksWorkflow()
+        {
+            _booksRepository = new BooksRepository();
+            _salesRepository = new SalesRepository();
+        }
+
+        public bool AddEntity(BookDto dto)
+        {
+            var existBook = _booksRepository.Get(dto.Title, dto.Author);
+            if (existBook == null)
+            {
+                _booksRepository.Add(new Book { Title = dto.Title, Author = dto.Author });
+            }      
+            
+            return true;
+        }
+
+        public bool DeleteEntity(int id)
+        {
+            return (_booksRepository.Remove(id) == true || _salesRepository.Remove(id) == true);
```

Comment:
_booksRepository.Remove(id) - it is OK
 || _salesRepository.Remove(id) - why do you use logical operator || here ?
 _salesRepository.Remove(id) - there can be a lot of rows in Sale table where BookId == 4 , but Id of sale can be 4,7,567,56676 egt - how to delete all these rows if they have the same BookId ?

Reply:
done